htlcswitch/mailbox.go

```diff
@@ -382,6 +390,14 @@ func (m *memoryMailBox) mailCourier(cType courierType) {
 		case pktCourier:
 			select {
 			case m.pktOutbox <- nextPkt:
+				m.pktCond.L.Lock()
+				// Only advance the pktHead if this packet
+				// is still at the head of the queue.
+				if m.pktHead != nil && m.pktHead == nextPktEl {
+					m.pktHead = m.pktHead.Next()
```

Comment:
If this packet is at the head of the queue, then isn't this the same as just setting the packet head to nil here?

Reply:
that's only true if the head is the last item in the queue, but there could be more